watch.go

```diff
@@ -70,11 +69,43 @@ func (fb *Firebase) Watch(notifications chan Event) error {
 			resp.Body.Close()
 		}()
 	scanning:
-		for scanner.Scan() {
+		for scanResult == nil {
 			// split event string
 			// 		event: put
 			// 		data: {"path":"/","data":{"foo":"bar"}}
-			parts := strings.Split(scanner.Text(), "\n")
+
+			var evt []byte
+			var dat []byte
+			isPrefix := true
+			var result []byte
+
+			// For possible results larger than 64 * 1024 bytes (MaxTokenSize)
+			// we need bufio#ReadLine()
+			// 1. step: scan for the 'event:' part. ReadLine() oes not return the \n
+			// so we have to add it to our result buffer.
+			evt, isPrefix, scanResult = scanner.ReadLine()
+			result = append(result, evt...)
+			result = append(result, '\n')
+
+			// 2. step: scan for the 'data:' part. Firebase returns just one 'data:'
+			// part, but the value can be very large. If we exceed a certain length
+			// isPrefix will be true until all data is read.
+			for {
+				dat, isPrefix, scanResult = scanner.ReadLine()
+				result = append(result, dat...)
+				if !isPrefix {
+					break
+				}
+			}
+			// Again we add the \n
+			result = append(result, '\n')
+			_, _, scanResult = scanner.ReadLine()
+			if scanResult != nil {
+				log.Println(scanResult)
```

Comment:
Don't understand what you mean. I you mean the for-loop in line#93: isPrefix will be false, in case of an error, too. We could check for scanResult != nil, but it doesn't change the semantics. I wanted to communicate the fact, that we have to call ReadLine in Line#94 for until isPrefix == false.